src/librustc/ty/mod.rs

```diff
@@ -1981,10 +1981,30 @@ impl<'a, 'gcx, 'tcx> AdtDef {
     }
 
     #[inline]
-    pub fn is_non_exhaustive(&self) -> bool {
+    fn is_non_exhaustive(&self) -> bool {
         self.flags.intersects(AdtFlags::IS_NON_EXHAUSTIVE)
     }
 
+    #[inline]
+    pub fn is_enum_non_exhaustive(&self) -> bool {
```

Comment:
I think we should call this method:

```rust
can_extend_variant_list
```

and add a suitable comment explaining how it is derived from a `#[non_exhaustive]` that appears on the enum (but not a variant).